include/util/json_renderer.hpp

```diff
@@ -64,131 +78,94 @@ struct Renderer
+}
 
-    void operator()(const String &string) const
-    {
-        out.push_back('\"');
-        const auto string_to_insert = escape_JSON(string.value);
-        out.insert(std::end(out), std::begin(string_to_insert), std::end(string_to_insert));
-        out.push_back('\"');
-    }
+template <> void Renderer<std::vector<char>>::write(const char *str)
+{
+    out.insert(out.end(), str, str + strlen(str));
+}
+template <> void Renderer<std::vector<char>>::write(char ch) { out.push_back(ch); }
 
-    void operator()(const Number &number) const
-    {
-        const std::string number_string = cast::to_string_with_precision(number.value);
-        out.insert(out.end(), number_string.begin(), number_string.end());
-    }
+template <> void Renderer<std::ostream>::write(const std::string &str) { out << str; }
 
-    void operator()(const Object &object) const
-    {
-        out.push_back('{');
-        for (auto it = object.values.begin(), end = object.values.end(); it != end;)
-        {
-            out.push_back('\"');
-            out.insert(out.end(), it->first.begin(), it->first.end());
-            out.push_back('\"');
-            out.push_back(':');
+template <> void Renderer<std::ostream>::write(const char *str) { out << str; }
 
-            mapbox::util::apply_visitor(ArrayRenderer(out), it->second);
-            if (++it != end)
-            {
-                out.push_back(',');
-            }
-        }
-        out.push_back('}');
-    }
+template <> void Renderer<std::ostream>::write(char ch) { out << ch; }
 
-    void operator()(const Array &array) const
-    {
-        out.push_back('[');
-        for (auto it = array.values.cbegin(), end = array.values.cend(); it != end;)
-        {
-            mapbox::util::apply_visitor(ArrayRenderer(out), *it);
-            if (++it != end)
-            {
-                out.push_back(',');
-            }
-        }
-        out.push_back(']');
-    }
+template <> void Renderer<std::string>::write(const std::string &str) { out += str; }
 
-    void operator()(const True &) const
-    {
-        const std::string temp("true");
-        out.insert(out.end(), temp.begin(), temp.end());
-    }
+template <> void Renderer<std::string>::write(const char *str) { out += str; }
 
-    void operator()(const False &) const
-    {
-        const std::string temp("false");
-        out.insert(out.end(), temp.begin(), temp.end());
-    }
-
-    void operator()(const Null &) const
-    {
-        const std::string temp("null");
-        out.insert(out.end(), temp.begin(), temp.end());
-    }
-
-  private:
-    std::vector<char> &out;
-};
+template <> void Renderer<std::string>::write(char ch) { out += ch; }
 
 inline void render(std::ostream &out, const Object &object)
 {
-    Value value = object;
-    mapbox::util::apply_visitor(Renderer(out), value);
+    Renderer renderer(out);
+    renderer(object);
+}
+
+inline void render(std::string &out, const Object &object)
```

Comment:
# 2.
Then I decided to add Renderer specialization for `std::string`(i.e. stopped using `std::stringstream` and started writing directly to `std::string`) and started using it in benchmark:
 
<img width="940" alt="string added" src="https://user-images.githubusercontent.com/266271/193301204-06e9d013-ddcc-4dcc-b398-729d03dea4e5.png">